### PR TITLE
feat: add C `ndarray` API and refactor `blas/ext/base/dssumors`

### PR DESCRIPTION
**Pull request by headlessNode**
## Description

> What is the purpose of this pull request?

This pull request:

-   add C `ndarray` API and refactor `blas/ext/base/dssumors`

## Related Issues

> Does this pull request have any related issues?

No. 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/dssumors/README.md [hunk 1]
```diff
@@ -146,6 +146,123 @@ console.log( v );
...
+```c
+#include "stdlib/blas/ext/base/dssumors.h"
+```
+
+#### stdlib_strided_dssumors( N, \*X, strideX )
+
+Computes the sum of single-precision floating-point strided array elements using ordinary recursive summation with extended accumulation and returning an extended precision result.
+
+```c
+const float x[] = { 1.0f, -2.0f, 0.0f/0.0f, 2.0f };
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dssumors/README.md]:
```suggestion
const float x[] = { 1.0f, 2.0f, 3.0f, 4.0f };
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumors/README.md [hunk 2]
```diff
@@ -146,6 +146,123 @@ console.log( v );
...
+```c
+double stdlib_strided_dssumors( const CBLAS_INT N, const float *X, const CBLAS_INT strideX );
+```
+
+#### stdlib_strided_dssumors_ndarray( N, \*X, strideX, offsetX )
+
+Computes the sum of single-precision floating-point strided array elements using ordinary recursive summation with extended accumulation and alternative indexing semantics and returning an extended precision result.
+
+```c
+const float x[] = { 1.0f, -2.0f, 0.0f/0.0f, 2.0f };
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dssumors/README.md]:
```suggestion
const float x[] = { 1.0f, 2.0f, 3.0f, 4.0f };
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumors/src/main.c [hunk 3]
```diff
@@ -17,31 +17,38 @@
...
+* @param X        input array
+* @param strideX  stride length
+* @return         output value
 */
-double stdlib_strided_dssumors( const int64_t N, const float *X, const int64_t stride ) {
+double API_SUFFIX(stdlib_strided_dssumors)( const CBLAS_INT N, const float *X, const CBLAS_INT strideX ) {
+	CBLAS_INT ox = stdlib_strided_stride2offset( N, strideX );
+	return API_SUFFIX(stdlib_strided_dssumors_ndarray)( N, X, strideX, ox );
+}
+
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dssumors/src/main.c:35]: missing JSDoc

File: lib/node_modules/@stdlib/blas/ext/base/dssumors/README.md [hunk 4]
```diff
@@ -146,6 +146,123 @@ console.log( v );
...
+    const float x[] = { 1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f, 7.0f, 8.0f };
+
+    // Specify the number of elements:
+    const int N = 5;
+
+    // Specify the stride length:
+    const int strideX = 2;
+
+    // Compute the sum:
+    double v = stdlib_strided_dsnansumors( N, x, strideX );
```
Comment (headlessNode) [inline lib/node_modules/@stdlib/blas/ext/base/dssumors/README.md]:
```suggestion
    double v = stdlib_strided_dssumors( N, x, strideX );
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumors/README.md [hunk 5]
```diff
@@ -146,6 +146,123 @@ console.log( v );
...
+
+#### stdlib_strided_dssumors( N, \*X, strideX )
+
+Computes the sum of single-precision floating-point strided array elements using ordinary recursive summation with extended accumulation and returning an extended precision result.
+
+```c
+const float x[] = { 1.0f, 2.0f, 3.0f, 4.0f };
+
+double v = stdlib_strided_dssumors( 4, x, 1 );
+// returns 1.0
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dssumors/README.md]:
```suggestion
// returns 10.0
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumors/README.md [hunk 6]
```diff
@@ -146,6 +146,123 @@ console.log( v );
...
+
+#### stdlib_strided_dssumors_ndarray( N, \*X, strideX, offsetX )
+
+Computes the sum of single-precision floating-point strided array elements using ordinary recursive summation with extended accumulation and alternative indexing semantics and returning an extended precision result.
+
+```c
+const float x[] = { 1.0f, 2.0f, 3.0f, 4.0f };
+
+double v = stdlib_strided_dssumors_ndarray( 4, x, 1, 0 );
+// returns 1.0
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dssumors/README.md]:
```suggestion
// returns 10.0
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumors/include/stdlib/blas/ext/base/dssumors.h [hunk 7]
```diff
@@ -31,7 +31,12 @@ extern "C" {
 /**
 * Computes the sum of single-precision floating-point strided array elements using  ordinary recursive summation with extended accumulation and returning an extended precision result.
 */
-double stdlib_strided_dssumors( const int64_t N, const float *X, const int64_t stride );
+double API_SUFFIX(stdlib_strided_dssumors)( const CBLAS_INT N, const float *X, const CBLAS_INT strideX );
+
+/**
+* Computes the sum of single-precision floating-point strided array elements using  ordinary recursive summation with extended accumulation and alternative indexing semantics and returning an extended precision result.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dssumors/include/stdlib/blas/ext/base/dssumors.h]:
```suggestion
* Computes the sum of single-precision floating-point strided array elements using ordinary recursive summation with extended accumulation and alternative indexing semantics and returning an extended precision result.
```

File: lib/node_modules/@stdlib/blas/ext/base/dssumors/include/stdlib/blas/ext/base/dssumors.h [hunk 8]
```diff
@@ -31,7 +31,12 @@ extern "C" {
 /**
 * Computes the sum of single-precision floating-point strided array elements using  ordinary recursive summation with extended accumulation and returning an extended precision result.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/dssumors/include/stdlib/blas/ext/base/dssumors.h]:
```suggestion
* Computes the sum of single-precision floating-point strided array elements using ordinary recursive summation with extended accumulation and returning an extended precision result.
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @headlessNode!

**Pull request merged**